Align tables columns in the machine details

anthonydillon opened the pull request:
## Done
Added bespoke styling to the tables in the machine details as maas-ui does. 

## QA
- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Go to the model details
- Check the tables match the tables visible in [the design](https://app.zeplin.io/project/5d2dfd8e5b67f14c5e1d2aef/screen/5e67bbca397a5b129eb35110)

## Details
Fixes https://github.com/canonical-web-and-design/jaas-dashboard/issues/424
Fixes https://github.com/canonical-web-and-design/jaas-dashboard/issues/426

## Screenshots
![Screenshot_2020-05-28 JAAS Dashboard](https://user-images.githubusercontent.com/1413534/83168893-1c042500-a10a-11ea-88e3-c6cd8288bb65.png)

hatched commented on src/scss/custom/_tables.scss:
```diff
@@ -14,3 +14,107 @@ th,
   min-width: 150px;
 }
+
+.model-details__apps {
```
I think that all of these changes should probably be in https://github.com/canonical-web-and-design/jaas-dashboard/blob/master/src/pages/Models/Details/_model-details.scss instead. This move should also resolve the stylelint issue.

barrymcgee commented:
The third and fourth columns of App and Unit tables are unreadable at <1400px

This is why I have resisted this approach thus far, it gets very messy, very quickly. 

<img width="1440" alt="Screenshot 2020-05-28 at 08 39 55" src="https://user-images.githubusercontent.com/505570/83113076-09b0c980-a0bf-11ea-8a62-f37cff5ff848.png">

spencerbygraves commented:
Guess it's a -1 based on the images above.

anthonydillon commented:
@spencerbygraves new screenshot attached to the description of the PR

spencerbygraves commented:
@anthonydillon thanks

anthonydillon commented:
@hatched @barrymcgee please do a good look through the different views as the styles were leaking out all over and I have refactored parts of the model listing.

Drive by's:
- Fixes the horizontal scrolling in the app
- Made the model-details__config component generic
- Add mixins the to at-rules for stylelint
- Moved the `p-filter` styling into a single place
- Made the model title a strong element instead of a div with bold styling
- Added message to the column headers in the relation table
- Replace the bespoke truncate class with the Vanilla one.

hatched commented:
Some tiny qa issues with the new truncated values.

The status column has the value "Unknown" truncated at 1679px.
<img width="458" alt="Screen Shot 2020-05-28 at 9 17 25 PM" src="https://user-images.githubusercontent.com/532033/83217327-b70bf780-a128-11ea-8fdd-5e1fce5b55fc.png">

IP addresses are truncated at the minimum width but there isn't a tooltip to get the actual value.
Statuses are truncated without an ellipsis and are also no tooltip.
<img width="686" alt="Screen Shot 2020-05-28 at 9 22 23 PM" src="https://user-images.githubusercontent.com/532033/83217637-72cd2700-a129-11ea-8cf6-dd1a1056ea2a.png">

Relation provider doesn't have a tooltip when truncated
<img width="579" alt="Screen Shot 2020-05-28 at 9 25 06 PM" src="https://user-images.githubusercontent.com/532033/83217787-c17ac100-a129-11ea-949f-55557e873f26.png">

Truncated unit names are missing tooltips
<img width="302" alt="Screen Shot 2020-05-28 at 9 27 25 PM" src="https://user-images.githubusercontent.com/532033/83217906-10285b00-a12a-11ea-9753-53e85794b540.png">

anthonydillon commented:
@hatched thanks, I noticed these too and plan to fix them in follow ups.
